### PR TITLE
refactor publishing charts - master

### PR DESCRIPTION
**Pull request by ljubon**
`gh-pages` branch PR: https://github.com/G-Research/charts/pull/76

The purpose of this PR is to perform a cleanup over `charts` repo and improve and automate the release process for new charts for all repos across G-Research organization

The file `.github/workflows/scripts/config.json` contains all info for repos and it will be used as a single source of truth for publishing charts.
By default, new packaged charts will be pushed to the `G-Research/charts:gh-pages` branch with an updated `index.yaml`

Current PR support publishing charts in two way:
- invoking a workflow from another repo (e.g G-Research/armada - example below)
   - each repo should have his own `APP_ID` and `APP_PRIVATE_KEY`, instruction for creating a GitHub app is given in file [GITHUB_APP_GUIDE.md](https://github.com/ljubon/charts/blob/master-public/GITHUB_APP_GUIDE.md)
- adding a Helm chart template structure to `master` branch  in a separate folder (e.g current example is `siembol`)

## TODO
- check/confirm which repos should start using new workflow and change accordingly `.github/workflows/scripts/config.json`

## Demo workflows and runs
The development branch is `INTERNAL-master` and `INTERNAL-refactor-gh-pages`
- Storm (ljubon/charts repo) [run ](https://github.com/ljubon/charts/actions/runs/2185612237)and [workflow](https://github.com/ljubon/charts/actions/runs/2185612237/workflow)
- Armada charts example 
    - ljubon/armada [workflow ](https://github.com/ljubon/armada/actions/runs/2185617524/workflow) (invoking push workflow)
    - ljubon/armada [run ](https://github.com/ljubon/armada/runs/6067198718?check_suite_focus=true)(invoking push workflow)
    - ljubon/charts [publishing charts](https://github.com/ljubon/charts/runs/6067202306?check_suite_focus=true)

## Example
Invoking a workflow from another repository:
```yaml
name: Release

on: 
  push:
    tags: 
      - 'v*'

jobs:
  invoke-push:
    uses: G-Research/charts/.github/workflows/invoke-push.yaml@master
    secrets:
      APP_ID: ${{ secrets.APP_ID }} 
      APP_PRIVATE_KEY: ${{ secrets.GH_APP_PEM }}
```

### CODE REVIEW

File: .github/workflows/scripts/push.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    await exec.exec('git', ['config', '--local', '--unset-all', 'http.https://github.com/.extraheader'], { cwd: checkoutPageDir })
+    await exec.exec('git', ['config', '--local', 'http.https://github.com/.extraheader', `AUTHORIZATION: basic ${headerToken_b64}`], { cwd: checkoutPageDir })
+    await exec.exec('git', ['config', '--local', 'user.name', 'G-Research charts'], { cwd: checkoutPageDir })
+    await exec.exec('git', ['config', '--local', 'user.email', 'charts@gr-oss.io'], { cwd: checkoutPageDir })
+
+    await exec.exec('helm', ['repo', 'index', '.'], { cwd: checkoutPageDir })
+    await exec.exec('git', ['add', 'index.yaml'], { cwd: checkoutPageDir })
+
+    await exec.exec('git', ['status'], { cwd: checkoutPageDir })
+    await exec.exec('git', ['commit', '-m', `Publish helm chart to ${context.payload.repository.owner.login}/${context.payload.repository.name}`, '--verbose'], { cwd: checkoutPageDir })
```
Comment (jayofdoom) [inline .github/workflows/scripts/push.js:27]: Suggestion: put more information in the commit message, perhaps about when/how the script was invoked?

File: GITHUB_APP_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+   - Organization https://github.com/organizations/ORG_NAME/settings/apps
+     - Replace `ORG_NAME` with the name of your organization
+   - Personal https://github.com/settings/apps
+
+2. Click to `New GitHub App`
+3. Set the name for your GitHub App (e.g. push-bot)
+4. As we don't host your app set `http://example.com` for `Homepage URL`
+5. Disable `Webhook` by unchecking `Active` button
+6. Change following permissions
```
Comment (jayofdoom) [inline GITHUB_APP_GUIDE.md:20]: Suggestion: screenshot of the properly-configured permissions page here would be nice.

**Review by jayofdoom — COMMENTED** [review]
This looks alright to me, but TBH I don't think I have enough context to approve it independently. A couple of suggestions inline, take them or leave them either is fine with me :)

File: GITHUB_APP_GUIDE.md (hunk 2 above)
Comment (ljubon) [reply GITHUB_APP_GUIDE.md:20]: I must say I agree with this, but this page is 50+ options and it's about 4-5 scrolling to get to the end
as you pointed to this specific part I've added additional info for all other options
update: also i found some syntax/typo issues which are fixed now

File: .github/workflows/scripts/push.js (hunk 1 above)
Comment (ljubon) [reply .github/workflows/scripts/push.js:27]: as this is the only way to publish this chart (after merging) I don't see a reason why in the commit message we would have `when/how` answer, but reasonably looking IMHO it was missing a `ref` a.k.a `version` of that chart which was published so in future in commit messages you can search explicit version
FYI I'm pushing that in the following commit

**Review by jgiannuzzi — APPROVED** [review]
Fantastic work, thanks @ljubon (and @pavlovic-ivan & @mkomatinovic!) 🚀

**Review by jgiannuzzi — DISMISSED** [review]

**Review by jgiannuzzi — APPROVED** [review]

**Pull request merged**